### PR TITLE
PXD-1384 feat(gulp): automate precompiling process

### PR DESCRIPTION
**Pull request by qingyashu**
- Use gulp for precompiling scrips
- Add readme and license
- Solve icon loading problem by adding postcss module

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Copyright 2017 University of Chicago
```
Comment (abgeorge7) [inline NOTICE]: needs to be updated to 2018 😝

**Review by abgeorge7 — COMMENTED** [review]
Do we still need the files in `dist` ? looks like they were removed
Also does removing the less loader affect importing base.less into data-portal at all?

File: NOTICE (hunk 1 above)
Comment (qingyashu) [reply NOTICE]: 😱

**Comment by qingyashu** [comment]
@abgeorge7  Seems `data-portal` has its own `less-loader` for all css even including files under `node_modules` so this won't break styles.
`dist` file still exist, but now stylesheets are precompiled so they are in one line, looks like being removed 😄

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    .pipe(gulpif(isDev, sourcemaps.write({sourceRoot: '/g3-ui/'})))
+    .pipe(gulp.dest('dist/components'))
+});
+
+gulp.task('icons', function() {
+  return gulp.src('src/images/icons/*')
+    .pipe(gulp.dest('dist/images/icons'))
+});
+
+// This is for old data portal base.less and will be deprecated
```
Comment (abgeorge7) [inline gulpfile.js]: is there a ticket for this? just so we don't forget

File: README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Clone the repo and try more components: 
+```
+git clone git@github.com:uc-cdis/gen3-ui-component.git
+npm run storybook
+```
+
+## For Developers
+Run following script before commit: 
+```
+npm run prepare
```
Comment (abgeorge7) [inline README.md:30]: do we want this anymore? or will gulp deal with it

**Review by abgeorge7 — COMMENTED** [review]
Looks good - just had minor comments and then once the tests are out of the dist folder I'll be good to go 👍

**Review by abgeorge7 — APPROVED** [review]

**Pull request merged**